pkg/apis/core/types_shoot.go

```diff
@@ -364,6 +364,8 @@ type DNSProvider struct {
 }
 
 // DNSIncludeExclude contains information about which domains shall be included/excluded.
+// Note: This struct is not used by Gardener anymore. However, providers might still be synced by the DNS extension which
+// is mainly why we can't deprecate or remove it, see https://github.com/gardener/gardener-extension-shoot-dns-service/blob/f599bec786ab6e27e6fd41d423d4e73dd4952f4f/pkg/admission/mutator/shoot_mutator.go#L110.
```

Comment:
What is the plan forward to drop it? Probably this DNS extension should adapt to use its provider config? cc @MartinWeindel 
Is there an issue which we can track and link here?

Reply:
The extension already uses its provider config. Due to compatibility reasons there is a provider sync functionality that copies `dns.providers` to the extension's provider config.
I wondered as well if we plan to drop this functionality. OTHO, we probably won't remove the `domains` and `zones` field anytime soon as it's part of the shoot API and would rather consider removing it when we promote the API to `v1`? 😄
We can however already add an deprecation note now?!

Reply:
> What is the plan forward to drop it? Probably this DNS extension should adapt to use its provider config? cc @MartinWeindel Is there an issue which we can track and link here?

Thanks @timuthy for your comments. I have not much to add to that. I see the Gardener core team in the driver seat for further planning to define what fields to drop with API `v1` as they are partially used by the DNSRecord controller, too.

Reply:
We can also drop fields in `v1beta1` after a proper deprecation period so that users have time to adopt. We already did so many times in the past.
Hence, I suggest to deprecate it know with the plan to finally remove after `v1.87` has been released (first release in 2024).
Then people have enough time to switch to `providerConfig` only.

Reply:
Fair enough, I adjusted the doc strings and release notes.

Reply:
FYI: https://github.com/gardener/gardener-extension-shoot-dns-service/issues/220